build_runner/lib/src/generate/options.dart

```diff
@@ -31,10 +33,14 @@ class BuildOptions {
   // For testing only, skips the build script updates check.
   bool skipBuildScriptCheck;
 
+  // For internal use only, flipped when a severe log occurred.
+  bool severeLogHandled = false;
```

Comment:
This runs counter to the goal in https://github.com/dart-lang/build/issues/95 and makes me a little nervous.

Any particular reason we are using this rather than throwing at the point of the log call?

Reply:
We can (and should) move the boolean to the BuildImpl class to remove the addition of dynamic state to BuildOptions. 

The logs are coming from a stream so unless that becomes synchronous  failing in the listener is sketchy.

Reply:
Do you still want me to land this with a TODO or hold?

Reply:
I'm fine with the change for now. We can revisit in the linked issue.